scripts/pre-commit/verify_migration_scripts.py

```diff
@@ -0,0 +1,182 @@
+import subprocess
+import sys
+import re
+import logging
+from pathlib import Path
+import typing as tp
+from datetime import datetime
+
+logger = logging.getLogger("verify_migration_script")
+logger.setLevel(logging.INFO)
+
+if not logger.handlers:
+    handler = logging.StreamHandler()
+    handler.setLevel(logging.INFO)
+    formatter = logging.Formatter("%(levelname)s - %(message)s")
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+CODE_EXTENSION = ".sql"
+BOOTSTRAP_FILE = "bootstrap.sql"
+METADATA_STORE_SCRIPTS_DIR = "metadata_store_scripts"
+
+
+class MigrationValidationError(Exception):
+    """Raised when a migration validation check fails."""
+
+
+def is_code_file(filepath: str) -> bool:
+    return Path(filepath).suffix == CODE_EXTENSION
+
+
+def get_project_root() -> Path:
+    project_root = subprocess.run(
+        ["git", "rev-parse", "--show-toplevel"],
+        capture_output=True,
+        text=True,
+        check=True
+    )
+    return Path(project_root.stdout.strip())
+
+
+def parse_date(version_str: str) -> datetime:
+    date_str = ".".join(version_str[1:].split(".")[:3])
+    return datetime.strptime(date_str, "%Y.%m.%d")
+
+
+def check_duplicate_migration_versions(new_migration_files: tp.List, old_migration_files: tp.List) -> None:
+    """
+    Check if there are any duplicate migration versions
+    1. Between new and old migration files.
+    2. Between new migration files.
+    """
+    new_versions = [file.split("_")[0] for file in new_migration_files if file != BOOTSTRAP_FILE]
+    old_versions = [file.split("_")[0] for file in old_migration_files]
+
+    duplicates = set(new_versions) & set(old_versions)
+    if duplicates:
+        raise MigrationValidationError(
+            f"Duplicate migration versions found: {duplicates}"
+        )
+
+    new_migration_files_set = set(new_migration_files)
+    duplicates = [file for file in new_migration_files_set if new_versions.count(file.split("_")[0]) > 1]
+    if duplicates:
```

Comment:
Can make use of walrus operator for one time throw away variables. 
```suggestion
    if duplicates := [file for file in new_migration_files_set if new_versions.count(file.split("_")[0]) > 1]
        raise ... 
```